src/librustc_mir/transform/const_prop.rs

```diff
@@ -781,7 +792,9 @@ impl<'tcx> Visitor<'tcx> for CanConstProp {
                         // end of the block anyway, and inside the block we overwrite previous
                         // states as applicable.
                         ConstPropMode::OnlyInsideOwnBlock => {}
-                        other => {
+                        ConstPropMode::NoPropagation => {}
+                        ConstPropMode::OnlyPropagateInto => {}
+                        other @ ConstPropMode::FullConstProp => {
```

Comment:
Part of exhaustively matching on things, but also part of the bugfix, because we don't go from `NoPropagation` to `OnlyPropagateInto` anymore.